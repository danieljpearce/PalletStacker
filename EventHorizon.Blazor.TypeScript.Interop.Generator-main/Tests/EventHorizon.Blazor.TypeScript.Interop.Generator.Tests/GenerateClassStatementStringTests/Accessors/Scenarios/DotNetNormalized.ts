declare module Examples {
    export class ExampleClass {
        get ref(): Vector3;
        get lock(): Vector3;
        get unlock(): Vector3;
        get object(): Vector3;
        get event(): Vector3;
        get bool(): Vector3;
        get virtual(): Vector3;
        get string(): Vector3;
        get this(): Vector3;
        get params(): Vector3;
    }
}