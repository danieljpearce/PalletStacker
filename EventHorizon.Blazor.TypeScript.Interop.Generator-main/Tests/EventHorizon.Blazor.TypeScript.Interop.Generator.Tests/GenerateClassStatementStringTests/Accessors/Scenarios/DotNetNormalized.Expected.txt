/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors
        private Vector3 __ref;
        public Vector3 @ref
        {
            get
            {
            if(__ref == null)
            {
                __ref = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "ref",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __ref;
            }
        }

        private Vector3 __lock;
        public Vector3 @lock
        {
            get
            {
            if(__lock == null)
            {
                __lock = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "lock",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __lock;
            }
        }

        private Vector3 __unlock;
        public Vector3 @unlock
        {
            get
            {
            if(__unlock == null)
            {
                __unlock = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "unlock",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __unlock;
            }
        }

        private Vector3 __object;
        public Vector3 @object
        {
            get
            {
            if(__object == null)
            {
                __object = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "object",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __object;
            }
        }

        private Vector3 __event;
        public Vector3 @event
        {
            get
            {
            if(__event == null)
            {
                __event = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "event",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __event;
            }
        }

        private Vector3 __bool;
        public Vector3 @bool
        {
            get
            {
            if(__bool == null)
            {
                __bool = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "bool",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __bool;
            }
        }

        private Vector3 __virtual;
        public Vector3 @virtual
        {
            get
            {
            if(__virtual == null)
            {
                __virtual = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "virtual",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __virtual;
            }
        }

        private Vector3 __string;
        public Vector3 @string
        {
            get
            {
            if(__string == null)
            {
                __string = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "string",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __string;
            }
        }

        private Vector3 __this;
        public Vector3 @this
        {
            get
            {
            if(__this == null)
            {
                __this = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "this",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __this;
            }
        }

        private Vector3 __params;
        public Vector3 @params
        {
            get
            {
            if(__params == null)
            {
                __params = EventHorizonBlazorInterop.GetClass<Vector3>(
                    this.___guid,
                    "params",
                    (entity) =>
                    {
                        return new Vector3() { ___guid = entity.___guid };
                    }
                );
            }
            return __params;
            }
        }
        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}